drivers/eks/eks_driver.go

```diff
@@ -1105,3 +1110,40 @@ func (d *Driver) waitForClusterUpdateReady(ctx context.Context, svc *eks.EKS, st
 
 	return nil
 }
+
+func getAMIs(ctx context.Context, ec2svc *ec2.EC2, state state) string {
+	version := state.KubernetesVersion
+	output, err := ec2svc.DescribeImagesWithContext(ctx, &ec2.DescribeImagesInput{
+		Filters: []*ec2.Filter{
+			&ec2.Filter{
+				Name:   aws.String("name"),
+				Values: aws.StringSlice([]string{fmt.Sprintf("%s%s", amiNamePrefix, version)}),
+			},
+		},
+	})
+	if err != nil {
+		logrus.WithError(err).Warn("getting AMIs from aws error")
+		return getLocalAMI(state)
+	}
+	rtnImage := ""
+	for _, image := range output.Images {
+		if *image.State != "available" && *image.ImageType != "machine" && !*image.Public {
```

Comment:
should these possibly be ORs ( || ) instead of ANDs ( && )? Seems like if any of those conditions are not met, we would want to consider the eimage